apps/prairielearn/src/ee/lib/contextEmbeddings.ts

```diff
@@ -0,0 +1,60 @@
+import { OpenAI } from 'openai';
+
+import { loadSqlEquiv, queryRows, queryOptionalRow } from '@prairielearn/postgres';
+
+import { QuestionGenerationContextEmbeddingSchema } from '../../lib/db-types.js';
+import { ServerJob } from '../../lib/server-jobs.js';
+
+import { DocumentChunk } from './context-parsers/documentation.js';
+
+const sql = loadSqlEquiv(import.meta.url);
+
+export function vectorToString(vec: number[]) {
```

Comment:
Please add JSDoc comments for all functions in this file.

Reply:
Done in ff0b8fae364583747e838607ffc674c57e963e85.